domain/build.gradle

```diff
@@ -0,0 +1,77 @@
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
+
+apply from: '../config/jacoco.gradle'
+
+def globalConfig = rootProject.extensions.getByName("ext")
+android {
+    compileSdkVersion globalConfig.getAt("android_compile_sdk_version")
+    defaultConfig {
+        minSdkVersion globalConfig.getAt("android_min_sdk_version")
+        targetSdkVersion globalConfig.getAt("android_target_sdk_version")
+        versionCode globalConfig.getAt("android_version_code")
+        versionName globalConfig.getAt("android_version_name")
+    }
+
+    flavorDimensions "stage"
+    productFlavors {
+        production {
+            dimension "stage"
+        }
+
+        staging {
+            dimension "stage"
+        }
+
+        development {
+            dimension "dev"
+        }
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled true
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+        debug {
+            minifyEnabled false
+        }
+    }
+
+    sourceSets {
+        test {
+            resources.srcDirs += ['src/test/resources']
+        }
+    }
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+
+    androidExtensions {
+        experimental = true
+    }
+}
+
+dependencies {
+    api project(':data')
```

Comment:
Any reason why this needs to be `api` instead of `implementation`?

Reply:
@Tuubz to allow app module access to data module components
so instead of 
```
app
    implementation project(':common-rx')
    implementation project(':data')
    implementation project(':domain')

domain
    implementation project(':data')
```
we can define
```
app
    implementation project(':common-rx')
    implementation project(':domain')

domain
    api project(':data')
```